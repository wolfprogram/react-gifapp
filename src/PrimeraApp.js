//import { Fragment } from "react";
//import React from 'react'
import propTypes from "prop-types";

const PrimeraApp = ({ saludo, subtitule }) => {
//  const PrimeraApp = ({ saludo = "Valor por default si no se envian nada" }) => {

  // const saludo = "Hola Mundo Const";

  // console.log(saludo, subtitule);
  return (
    <>
      <h1>Hola Mundo</h1>
      <h1>{saludo}!!!</h1>
      <p>{subtitule}</p>
    </>
  );
};

PrimeraApp.propTypes = {
  saludo: propTypes.string.isRequired
};

PrimeraApp.defaultProps = {
  subtitule: 'Soy un subtitule'
}

export default PrimeraApp;