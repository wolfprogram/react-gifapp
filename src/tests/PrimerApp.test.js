
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'
import PrimeraApp from "../PrimeraApp"





describe('Prueba de <PrimerApp />', () => { 
    test('debe de mostrar <PrimeraApp /> correctamente', () => { 
         const saludo = 'Saludo requerido';
         const wrapper = shallow( <PrimeraApp saludo = {saludo} />);

         expect( wrapper ).toMatchSnapshot();

        //  createSerialazer();

     });

     test('debe de mostar el subtitulo enviado por props', () => { 
         const saludo = 'Saludo en prueba subtra';
         const subtitulo = 'Soy un subtitule';

         const wrapper = shallow( <PrimeraApp 
            saludo = {saludo}
            subtitulo = {subtitulo}
             />);

         const textoParrafo = wrapper.find('p').first().text();
        //  const textoParrafo = wrapper.find('p').text();
         console.log(textoParrafo);

         expect( textoParrafo).toBe(subtitulo)

      });
 })

