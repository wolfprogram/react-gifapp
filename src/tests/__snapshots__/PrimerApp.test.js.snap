// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <PrimerApp /> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <h1>
    Saludo requerido
    !!!
  </h1>
  <p>
    Soy un subtitule
  </p>
</Fragment>
`;
